app/api/listings/route.ts

```
@@ -1,0 +1,317 @@
+ * Handles GET requests to retrieve listing data.
+ *
+ * If an `id` query parameter is provided, returns a single listing by ID or a 404 error if not found. Without an `id`, returns a paginated list of listings with total count and pagination metadata. Each listing includes additional mocked fields for distance, rating, and reviews.
+ *
+ * @returns A JSON response containing either a single formatted listing or a paginated list of listings with metadata.
+ */
+export async function GET(request: Request) {
+  const supabase = await getSupabaseRouteHandler();
+  const { searchParams } = new URL(request.url);
+
+  const id = searchParams.get("id");
+  const page = parseInt(searchParams.get("page") || "1", 10);
```

Biome (1.9.4)

[error] Use Number.parseInt instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseInt instead.


(lint/style/useNumberNamespace)